runtime-testsuite/src/org/antlr/v4/testgen/TestGenerator.java

```diff
@@ -30,202 +30,315 @@
+
+    // set during genTarget()
+    protected String targetName;
+    protected String lineSeparator = System.getProperty("line.separator");
+    protected ErrorBuffer errorBuffer = new ErrorBuffer();
+
 	// This project uses UTF-8, but the plugin might be used in another project
 	// which is not. Always load templates with UTF-8, but write using the
 	// specified encoding.
-	protected final String encoding;
-	protected final String targetName;
-	protected final String rootDir;
-	protected final File outputDir;
-	protected final File testTemplates;
-	protected final File runtimeTemplate;
-	protected final boolean visualize;
-
-	public TestGenerator(String encoding, String targetName, String rootDir, File outputDir, File testTemplates, File runtimeTemplate, boolean visualize) {
-		this.encoding = encoding;
-		this.targetName = targetName;
-		this.rootDir = rootDir;
-		this.outputDir = outputDir;
-		this.testTemplates = testTemplates;
-		this.runtimeTemplate = runtimeTemplate;
-		this.visualize = visualize;
-	}
 
-	private String getTargetName() {
-		return targetName;
-	}
+    /**
+     * Consume a command line option.
+     * 
+     * @param arg       the argument to examine
+     * @param moreArgs  The rest of the arguments.  If arg is recognized as an
+     *                  option that requires a value, the value is consumed from
+     *                  this iterator; TestGenException is thrown if not enough.
+     * @return true if option was recognized and consumed; false if unrecognized
+     * @throws TestGenException
+     */
+    public boolean acceptOption(String arg, Iterator<String> moreArgs) {
+        try {
+            if ("-root".equals(arg))
+                rootDir = new File(moreArgs.next());
+            else if ("-outdir".equals(arg))
+                outDir = new File(moreArgs.next());
+            else if ("-templates".equals(arg))
+                testTemplatesRoot = new File(moreArgs.next());
+            else if ("-encoding".equals(arg))
+                encoding = moreArgs.next();
+            else if ("viz".equals(arg))
+                visualize = true;
+            else if ("-browsers".equals(arg))
+                browsers = true;
+            else if ("-target".equals(arg))
+                acceptOptionMulti(targets, arg, moreArgs);
+            else if ("-exclude".equals(arg))
+                acceptOptionMulti(excludes, arg, moreArgs);
+            else
+                return false;   // option not recognized
+            return true;        // ok, consumed
+        } catch (NoSuchElementException ex) {
+            throw new TestGenException("Option requires a value: %s", arg);
+        }
+    }
+
+    /** Consume one or more args, each arg but the last ending with a comma. */
+    protected static void acceptOptionMulti(Collection<String> out, String arg, Iterator<String> moreArgs) {
+        try {
+            String v = moreArgs.next();
+            while (v.endsWith(",")) {
```

Comment:
❗️ This does not seem like the expected command line argument handling. As a user, I would expect one of the following to work:

* Passing the same command line argument with the flag multiple times, e.g. `-exclude A -exclude B`. This appears to work based on what I see in the code review.
* Passing the command line once, but having one comma-separated value, e.g. `-exclude A,B`. If a space were allowed, I would expect the required form to be `-exclude "A, B"` (quotes used to make one argument that contains a space). Neither of these forms currently works.

Assuming the first of these forms really does work, my recommendation is simply to remove the handling for `,` and update the comment for the method.